compiler/rustc_session/src/session.rs

```diff
@@ -950,10 +950,10 @@ fn default_emitter(
         t => t,
     };
     match sopts.error_format {
-        config::ErrorOutputType::HumanReadable(kind) => {
-            let (short, color_config) = kind.unzip();
+        config::ErrorOutputType::HumanReadable(kind, color_config) => {
+            let short = matches!(kind, HumanReadableErrorType::Short);
```

Comment:
Nit: very tiny nit: this `matches!(kind, HumanReadableErrorType::Short)` is repeated in quite a few places. Since `HumanReadableErrorType` derives `PartialEq`, this could just be

```rs
kind == HumanReadableErrorType::Short
```

or maybe if we want the usage site to look like `kind.short()`

```rs
impl HumanReadableErrorType {
    fn short(self) -> bool { self == HumanReadableErrorType::Short }
}
```